### PR TITLE
Bring over cherry picks from camel-spring-boot-examples-4.4.0

### PR DESCRIPTION
**Pull request by cunningt**
Went through all the commits between camel-spring-boot-examples-4.0.0 and camel-spring-boot-examples-4.4.0 and brought over cherry picks for every commit that applied to a supported example.

### CODE REVIEW

**Comment by mcarlett** [comment]
@cunningt do we change the BOMs with the product BOM in a new commit after this PR?

File: activemq/pom.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        <category>Messaging</category>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+	<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <activemq-version>5.18.3</activemq-version>
+    </properties>
+
+    <dependencyManagement>
+        <dependencies>
+            <!-- Spring Boot BOM -->
+            <dependency>
```
Comment (Croway) [inline activemq/pom.xml:44]: platform bom should be used here

File: resilience4j/service2/src/main/resources/application.properties [hunk 2]
```diff
@@ -15,10 +15,14 @@
...
 ## ---------------------------------------------------------------------------
 
-# the name of Camel
-camel.springboot.name=Service2
-camel.springboot.main-run-controller=true
-
-# the port number the service will use for accepting incoming HTTP requests
-service2.port=7070
-
+- routeConfiguration:
```
Comment (Croway) [inline resilience4j/service2/src/main/resources/application.properties]: yaml in a properties file?

File: servicecall/services/src/main/resources/application.properties [hunk 3]
```diff
@@ -15,15 +15,19 @@
...
-# Camel
-camel.springboot.main-run-controller=true
-camel.springboot.jmx-enabled=false
-
-# expose actuator endpoint via HTTP
-management.endpoints.web.exposure.include=camelroutes,jolokia,metrics,prometheus,health
-# turn on actuator health check
-management.endpoint.health.enabled = true
\ No newline at end of file
+- route:
```
Comment (Croway) [inline servicecall/services/src/main/resources/application.properties]: yaml in a properties file?

File: activemq/pom.xml (hunk 1 above)
Comment (cunningt) [reply activemq/pom.xml:44]: Fixed (using platform bom), checked other projects for similar cases

File: resilience4j/service2/src/main/resources/application.properties (hunk 2 above)
Comment (cunningt) [reply resilience4j/service2/src/main/resources/application.properties]: Fixed, reverted to previous application.properties file

File: servicecall/services/src/main/resources/application.properties (hunk 3 above)
Comment (cunningt) [reply servicecall/services/src/main/resources/application.properties]: Fixed, reverted to previous application.properties file

**Comment by cunningt** [comment]
@mcarlett @Croway Thanks for the feedback and catching the merge errors - application.properties and BOMs now fixed

File: activemq/pom.xml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    <dependencies>
+        <!-- Spring Boot -->
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-actuator</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
```
Comment (mcarlett) [inline activemq/pom.xml]: we should use undertow instead of tomcat

File: amqp/pom.xml [hunk 5]
```diff
@@ -56,32 +56,21 @@
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
-            <exclusions>
-                <exclusion>
-                    <groupId>org.springframework.boot</groupId>
-                    <artifactId>spring-boot-starter-tomcat</artifactId>
-                </exclusion>
-            </exclusions>
         </dependency>
```
Comment (mcarlett) [inline amqp/pom.xml]: don't we use undertow in downstream?

**Review by mcarlett — COMMENTED** [review]
I see you replace tomcat with undertow, I suppose in downstream we usually have undertow by default

**Comment by Croway** [comment]
Hi @cunningt the following component is missing https://github.com/apache/camel-spring-boot-examples/tree/main/artemis and it seems like that some pom.xml have a regression, (removal of undertow and the tomcat exclusion)

**Comment by cunningt** [comment]
> Hi @cunningt the following component is missing https://github.com/apache/camel-spring-boot-examples/tree/main/artemis and it seems like that some pom.xml have a regression, (removal of undertow and the tomcat exclusion)

My intention here was only to cherry pick over changes (like the one Torsten brought up) that apply to current quickstarts in the branch - not bring in new quickstarts - that's a different conversation I think.

**Comment by Croway** [comment]
Ohhh okok! sorry for confusion, yet, I still think that we should use downstream best practices (undertow, platform bot, omp..)

**Review by mcarlett — APPROVED** [review]

**Review by Croway — APPROVED** [review]

**Pull request merged**